src/fileprovider.cs

```diff
@@ -0,0 +1,301 @@
+using XamCore.Foundation;
+
+namespace XamCore.FileProvider {
+
+	[iOS (11,0)]
+	[ErrorDomain ("NSFileProviderErrorDomain")]
+	[Native]
+	public enum NSFileProviderError : nint {
+		NotAuthenticated = -1000,
+		FilenameCollision = -1001,
+		SyncAnchorExpired = -1002,
+		InsufficientQuota = -1003,
+		ServerUnreachable = -1004,
+		NoSuchItem = -1005
+	}
+
+	[iOS (11,0)]
+	enum NSFileProviderItemIdentifier {
+		// Lets follow swift names
+		[Field ("NSFileProviderRootContainerItemIdentifier")]
+		RootContainer,
+
+		[Field ("NSFileProviderWorkingSetContainerItemIdentifier")]
+		WorkingSet,
+
+		[Field ("NSFileProviderAllDirectoriesContainerItemIdentifier")]
+		AllDirectories,
+	}
+
+	[iOS (11,0)]
+	[Native]
+	[Flags]
+	public enum NSFileProviderItemCapabilities : nuint {
+		Reading = 1 << 0,
+		Writing = 1 << 1,
+		Reparenting = 1 << 2,
+		Renaming = 1 << 3,
+		Trashing = 1 << 4,
+		Deleting = 1 << 5,
+		AddingSubItems = Writing,
+		ContentEnumerating = Reading,
+		All = Reading | Writing | Reparenting | Renaming | Trashing | Deleting
+	}
+
+	[iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface NSFileProviderDomain {
+
+		[Export ("initWithIdentifier:displayName:pathRelativeToDocumentStorage:")]
+		IntPtr Constructor (string identifier, string displayName, string pathRelativeToDocumentStorage);
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("displayName")]
+		string DisplayName { get; }
+
+		[Export ("pathRelativeToDocumentStorage")]
+		string PathRelativeToDocumentStorage { get; }
+	}
+
+	interface INSFileProviderEnumerationObserver { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderEnumerationObserver {
+
+		[Abstract]
+		[Export ("didEnumerateItems:")]
+		void DidEnumerateItems (INSFileProviderItem [] updatedItems);
+
+		[Abstract]
+		[Export ("finishEnumeratingUpToPage:")]
+		void FinishEnumerating ([NullAllowed] NSData nextPage);
+
+		[Abstract]
+		[Export ("finishEnumeratingWithError:")]
+		void FinishEnumerating (NSError error);
+	}
+
+	interface INSFileProviderChangeObserver { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderChangeObserver {
+
+		[Abstract]
+		[Export ("didUpdateItems:")]
+		void DidUpdateItems (INSFileProviderItem [] updatedItems);
+
+		[Abstract]
+		[Export ("didDeleteItemsWithIdentifiers:")]
+		void DidDeleteItems (string [] deletedItemIdentifiers);
+
+		[Abstract]
+		[Export ("finishEnumeratingChangesUpToSyncAnchor:moreComing:")]
+		void FinishEnumeratingChanges (NSData anchor, bool moreComing);
+
+		[Abstract]
+		[Export ("finishEnumeratingWithError:")]
+		void FinishEnumerating (NSError error);
+	}
+
+	interface INSFileProviderEnumerator { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderEnumerator {
+
+		[Abstract]
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Abstract]
+		[Export ("enumerateItemsForObserver:startingAtPage:")]
+		void EnumerateItems (INSFileProviderEnumerationObserver observer, NSData page);
```

Comment:
I think it would be better to name the second parameter `startingAtPage`, or `startPage`, which conveys more information.